superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -56,6 +55,21 @@ const EXPLORE_CHART_DEFAULT = {
   viz_type: 'table',
 };
 
+enum LIMITING_FACTOR {
+  QUERY = 'QUERY',
+  QUERY_AND_DROPDOWN = 'QUERY_AND_DROPDOWN',
+  DROPDOWN = 'DROPDOWN',
+  NOT_LIMITED = 'NOT_LIMITED',
+}
+
+function limitReached(displayLimitReached: boolean) {
+  return displayLimitReached;
+}
+
+function defaultDropdown(queryLimit: number, limitingFactor: string) {
```

Comment:
can we make the name more indicative of what this function does? Does it return a boolean, then maybe `shouldShowDefaultDropdown`? or `shouldUse...` whatever fits better.